In [27]:
"""
Import packages 
"""
import os # to change working directory 
import numpy as np 
import pandas as pd # standard package for data analysis in Python
import matplotlib.pyplot as plt # base package for visulisations in Python
import seaborn as sns # visulisation opackage built additional on matplotlib with more features

# Note we can use hash (#) or 6 quotation marks """ """ for comments 

In [32]:
# Set working directory. The working directory is the folder you use for importing stuff into Jupyter. 


# os.getcwd() # Tells you what the current working directory is
# os.chdir('C:/Users/Administrator/Desktop/Projects/Titanic/Datasets') # Changes the working directory

In [29]:
"""
Import datasets 
"""

df = pd.read_csv('train.csv') # Imports the training set 
df1 = pd.read_csv('test.csv') # Imports the test set

In [ ]:
"""
Exploratory Data Analysis 
"""

df.head() # Pandas function to display the first 5 rows. We do this to have a rough overview of the data 

This is a markdown cell. It is basically for typing text instead of code. Executing this cell gives a text. 

The data dictionary is as follows: 

* survival : Survival (0 = No, 1 = Yes)
* pclass : Ticket class (1 = 1st, 2 = 2nd, 3 = 3rd)
* sex : Sex 	
* Age : Age in years 	
* sibsp : # of siblings / spouses aboard the Titanic 	
* parch : # of parents / children aboard the Titanic 	
* ticket : Ticket number 	
* fare 	: Passenger fare 	
* cabin : Cabin number 	
* embarked : Port of Embarkation (C = Cherbourg, Q = Queenstown, S = Southampton)

In [33]:
n_rows, n_cols = df.shape
print(f"Number of rows is {n_rows}")
print(f"Number of columns is {n_cols}")

Number of rows is 891
Number of columns is 12


In [18]:
df.info() # Tells us about the data type of each column 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [36]:
df.head(10) # Gives a statistical overview of each variable 

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

In [20]:
df.isnull().sum() # Check for missing values

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Missing values: 
* Age : 177 
* Cabin : 687 
* Embarked : 2

Using count(), find out the percentage of variables with missing data

In [34]:
round((1 - (df.count() / n_rows)) * 100,2)

PassengerId     0.00
Survived        0.00
Pclass          0.00
Name            0.00
Sex             0.00
Age            19.87
SibSp           0.00
Parch           0.00
Ticket          0.00
Fare            0.00
Cabin          77.10
Embarked        0.22
dtype: float64

In [21]:
df.nunique() # Tells us the number of unique values 

PassengerId    891
Survived         2
Pclass           3
Name           891
Sex              2
Age             88
SibSp            7
Parch            7
Ticket         681
Fare           248
Cabin          147
Embarked         3
dtype: int64